modules/hawkular-api-parent/hawkular-listener/src/main/java/org/hawkular/listener/bus/InventoryEventListener.java

```diff
@@ -0,0 +1,889 @@
+import org.hawkular.alerts.api.model.condition.ThresholdCondition.Operator;
+import org.hawkular.alerts.api.model.condition.ThresholdRangeCondition;
+import org.hawkular.alerts.api.model.dampening.Dampening;
+import org.hawkular.alerts.api.model.trigger.Mode;
+import org.hawkular.alerts.api.model.trigger.Trigger;
+import org.hawkular.alerts.api.services.DefinitionsService;
+import org.hawkular.inventory.api.model.Resource;
+import org.hawkular.inventory.api.model.ResourceType;
+import org.hawkular.inventory.bus.api.InventoryEvent;
+import org.hawkular.inventory.bus.api.InventoryEventMessageListener;
+import org.hawkular.inventory.bus.api.ResourceEvent;
+import org.hawkular.inventory.bus.api.ResourceTypeEvent;
+import org.jboss.logging.Logger;
+
+/**
+ * <p>
+ * An adapter that processes Hawkular Metrics data, extracts relevant metric datums, translates them to Alerting
+ * Data format, and forwards them for Alert processing.
+ * </p>
+ * This is useful only when deploying into the Hawkular Bus with Hawkular Metrics. The expected message payload should
+ * be JSON representation of {@link MetricDataMessage}.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@MessageDriven(messageListenerInterface = MessageListener.class, activationConfig = {
+        @ActivationConfigProperty(propertyName = "destinationType", propertyValue = "javax.jms.Topic"),
+        @ActivationConfigProperty(propertyName = "destination", propertyValue = "HawkularInventoryChanges") })
+@TransactionAttribute(value = TransactionAttributeType.NOT_SUPPORTED)
+public class InventoryEventListener extends InventoryEventMessageListener {
+    private final Logger log = Logger.getLogger(InventoryEventListener.class);
+
+    private static final String DEFINITIONS_SERVICE = "java:global/hawkular-alerts-rest/CassDefinitionsServiceImpl";
+
+    private static final long MINUTE = 60000L;
+
+    private InitialContext ctx;
+    private DefinitionsService definitions;
+
+    @Override
+    protected void onBasicMessage(InventoryEvent<?> event) {
+        switch (event.getAction()) {
+            case CREATED:
+            case DELETED:
+                if (event instanceof ResourceTypeEvent) {
+                    handleResourceTypeEvent((ResourceTypeEvent) event);
+                } else if (event instanceof ResourceEvent) {
+                    handleResourceEvent((ResourceEvent) event);
+                }
+                break;
+            default:
+                break; // not interesting
+        }
+    }
+
+    private void handleResourceTypeEvent(ResourceTypeEvent event) {
+        try {
+            init();
+
+            String tenantId = event.getTenant().getId();
+            ResourceType rt = event.getObject();
+            String type = rt.getId();
+            switch (type) {
+                case "Datasource": {
+                    // Available Connections
+                    String groupTriggerId = "DS_Conn";
```

Comment:
What about moving these lines of code to a separate method, there are lots of lines of code in this case block.